plugin/advisor/catalog/walk_through.go

```diff
@@ -116,11 +148,372 @@ func (d *databaseState) changeState(in tidbast.StmtNode) error {
 		return d.createTable(node)
 	case *tidbast.DropTableStmt:
 		return d.dropTable(node)
+	case *tidbast.AlterTableStmt:
+		return d.alterTable(node)
 	default:
 		return nil
 	}
 }
 
+func (d *databaseState) alterTable(node *tidbast.AlterTableStmt) error {
+	if node.Table.Schema.O != "" && node.Table.Schema.O != d.name {
+		return &WalkThroughError{
+			Type:    ErrorTypeAccessOtherDatabase,
+			Content: fmt.Sprintf("Database `%s` is not the current database `%s`", node.Table.Schema.O, d.name),
+		}
+	}
+
+	schema, exists := d.schemaSet[""]
+	if !exists {
+		schema = d.createSchema("")
+	}
+
+	table, exists := schema.tableSet[node.Table.Name.O]
+	if !exists {
+		if !schema.context.CheckIntegrity {
+			return nil
+		}
+
+		return &WalkThroughError{
+			Type:    ErrorTypeTableNotExists,
+			Content: fmt.Sprintf("Table `%s` does not exist", node.Table.Name.O),
+		}
+	}
+
+	for _, spec := range node.Specs {
+		switch spec.Tp {
+		case tidbast.AlterTableOption:
+			for _, option := range spec.Options {
+				switch option.Tp {
+				case tidbast.TableOptionCollate:
+					table.collation = option.StrValue
+				case tidbast.TableOptionComment:
+					table.comment = option.StrValue
+				case tidbast.TableOptionEngine:
+					table.engine = option.StrValue
+				}
+			}
+		case tidbast.AlterTableAddColumns:
+			for _, column := range spec.NewColumns {
+				pos := len(table.columnSet)
+				if spec.Position != nil && len(spec.NewColumns) == 1 {
+					var err error
+					pos, err = table.reorderColumn(spec.Position)
+					if err != nil {
+						return err
+					}
+				}
+				if err := table.createColumn(column, pos); err != nil {
+					return err
+				}
+			}
+			// MySQL can add table constraints in ALTER TABLE ADD COLUMN statements.
+			for _, constraint := range spec.NewConstraints {
+				if err := table.createConstraint(constraint); err != nil {
+					return err
+				}
+			}
+		case tidbast.AlterTableAddConstraint:
+			if err := table.createConstraint(spec.Constraint); err != nil {
+				return err
+			}
+		case tidbast.AlterTableDropColumn:
+			if err := table.dropColumn(spec.OldColumnName.Name.O); err != nil {
+				return err
+			}
+		case tidbast.AlterTableDropPrimaryKey:
+			if err := table.dropIndex(PrimaryKeyName); err != nil {
+				return err
+			}
+		case tidbast.AlterTableDropIndex:
+			if err := table.dropIndex(spec.Name); err != nil {
+				return err
+			}
+		case tidbast.AlterTableDropForeignKey:
+			// we do not deal with DROP FOREIGN KEY statements.
+		case tidbast.AlterTableModifyColumn:
+			if err := table.modifyColumn(spec.NewColumns[0], spec.Position); err != nil {
+				return err
+			}
+		case tidbast.AlterTableChangeColumn:
+			if err := table.changeColumn(spec.OldColumnName.Name.O, spec.NewColumns[0], spec.Position); err != nil {
```

Comment:
We have modifyColumn, changeColumn, alterColumn here. Could we make them consistent?